onlinejudge/service/atcoder.py

```diff
@@ -642,44 +642,51 @@ def _get_tag_lang(cls, tag: bs4.Tag) -> Optional[str]:
     def _find_sample_tags(cls, soup: bs4.BeautifulSoup) -> Iterator[Tuple[bs4.Tag, bs4.Tag]]:
         expected_strings = ('入力例', '出力例', 'Sample Input', 'Sample Output')
 
-        def get_header(tag, expected_tag_name):
+        def get_header(tag, expected_tag_name, ignore_tag_names):
+            assert not (expected_tag_name in ignore_tag_names)
+            while tag and tag.name in ignore_tag_names:
+                tag = tag.find_previous_sibling()
             if tag and tag.name == expected_tag_name and tag.string and any(s in tag.string for s in expected_strings):
                 return tag
             return None
 
         for pre in soup.find(id='task-statement').find_all('pre'):
             log.debug('pre tag: %s', str(pre))
 
-            # the standard format: #task-statement h3+pre
-            # used by AtCoder's JavaScript, sometimes used with .prettyprint
-            # example: https://atcoder.jp/contests/abc114/tasks/abc114_d
-            # NOTE: The AtCoder's JavaScript (at https://atcoder.jp/public/js/contest.js?v=201911110917 version) supports:
-            #     -   "#task-statement h3+pre" format for Copy buttons of <h3> and <pre> tags
-            #     -   "pre.prettyprint" format for Copy buttons of <pre> tags
-            h3 = get_header(tag=pre.find_previous_sibling(), expected_tag_name='h3')
-            if h3:
-                yield (pre, h3)
-                continue
+            # a very old format: #task-statement p+pre.literal-block
+            # p+(br)+pre.literal-block, p+(br)+(br)+pre.literal-block, ... is also accepted
+            # entirely unsupported by AtCoder's JavaScript
+            # NOTE: h3+p+pre.literal-block can be regarded as the standard format, so FIRSTLY check a very old format.
+            # example: https://atcoder.jp/contests/utpc2011/tasks/utpc2011_1
+            if 'literal-block' in pre.attrs.get('class', []):
+                p = get_header(tag=pre.find_previous_sibling(), expected_tag_name='p', ignore_tag_names=('br'))
```

Comment:
```suggestion
                p = get_header(tag=pre.find_previous_sibling(), expected_tag_name='p', ignore_tag_names=['br'])
```

これ `['b', 'r']` 相当の挙動をしてるように見えます。

Reply:
これはかなりヤバかったです、ありがとうございます。
僕はこれを直すのに['br']ではなく('br',)と1要素のタプルを作るほうが良いと感じますが、カンマ一つで挙動が変わるのは分かりにくいかもと懸念します。

Reply:
分かりにくいかもという懸念はもっともですが、`(hoge,)` は Python としては頻繁に使われるので気にしなくてよいと考えています。